flink-tests/src/test/java/org/apache/flink/test/checkpointing/ChangelogPeriodicMaterializationSwitchStateBackendITCase.java

```diff
@@ -91,39 +94,50 @@ public void testSwitchFromDisablingToEnablingInClaimMode() throws Exception {
         MiniCluster miniCluster = cluster.getMiniCluster();
         StreamExecutionEnvironment env1 =
                 getEnv(delegatedStateBackend, firstCheckpointFolder, false, 100, 600000);
-        JobGraph firstJobGraph = buildJobGraph(env1);
+        SharedReference<AtomicInteger> currentCheckpointNum =
+                sharedObjects.add(new AtomicInteger(0));
+        JobGraph firstJobGraph =
+                buildJobGraph(env1, TOTAL_ELEMENTS / 5, TOTAL_ELEMENTS / 4, currentCheckpointNum);
+
+        try {
+            miniCluster.submitJob(firstJobGraph).get();
+            miniCluster.requestJobResult(firstJobGraph.getJobID()).get();
+        } catch (Exception ex) {
+            Preconditions.checkState(
+                    ExceptionUtils.findThrowable(ex, ArtificialFailure.class).isPresent());
+        }
 
-        miniCluster.submitJob(firstJobGraph).get();
-        waitForAllTaskRunning(miniCluster, firstJobGraph.getJobID(), true);
-        miniCluster.triggerCheckpoint(firstJobGraph.getJobID()).get();
-        miniCluster.cancelJob(firstJobGraph.getJobID()).get();
         String firstRestorePath =
                 getLatestCompletedCheckpointPath(firstJobGraph.getJobID(), miniCluster).get();
 
-        // 1st restore, switch from disable to enable
+        // 1st restore, switch from disable to enable. Make materializationInterval = 600000, to
```

Comment:
Actually, We could set materializationInterval to a negative value to avoid materialization currently.
See the description of `StateChangelogOptions#PERIODIC_MATERIALIZATION_INTERVAL`.